hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -67,9 +67,8 @@ public void preHandle(@NonNull final PreHandleContext context) {
             final var key = asHederaKey(op.adminKeyOrThrow());
             key.ifPresent(context::addToReqNonPayerKeys);
         }
-        // TODO This code isn't right, autoRenewAccountId may be null
-        if (op.hasAutoRenewAccountId() && !op.autoRenewAccountId().equals(AccountID.DEFAULT)) {
-            context.addNonPayerKey(op.autoRenewAccountId(), INVALID_AUTORENEW_ACCOUNT);
+        if (op.hasAutoRenewAccountId() && !op.autoRenewAccountIdOrThrow().equals(AccountID.DEFAULT)) {
```

Comment:
I think in this case we should not do `op.autoRenewAccountIdOrThrow()`. We will be removing autorenew account if it is set explicitly as `AccountID.DEFAULT`.

It should be `!op.autoRenewAccountId().equals(AccountID.DEFAULT)`.

Reply:
That's what it was, but it had a warning that `autoRenewaccountId` could be null. Should it be `!op.autoRenewAccountIdOrElse(AccountID.DEFAULT).equals(AccountID.DEFAULT)`?